[#1891] Update Cypress to v10

vvidday opened the pull request:
<!--
    If this pull request fully addresses an issue, use:
    Fixes #xxxx

    If this pull request partially addresses an issue, use:
    Part of #xxxx

    If this pull request addresses multiple issues, put them in multiple lines:
    Fixes #xxxx
    Fixes #yyyy

    Format the title of the pull request with most relevant issue number as follows:
    [#xxxx] Title
-->
Part of #1891

### Proposed commit message
<!--
    Propose a detailed commit message for this pull request within the triple backticks below.
    Wrap lines at 72 characters.

    Guide on how to write a good commit message:
    https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message
-->
```
Currently, we are using Cypress 9, which is 3 major versions behind 
the latest release. Older versions are less likely to be supported and
maintained.

Let’s update Cypress to version 10.
```

### Other information
<!--
    Are there other relevant information, such as special testing instructions, 
    which will help the reviewer better understand the code?

    You may also include a brief description of why the problem occurred.
-->
Test files are renamed according to the [new recommended spec pattern](https://docs.cypress.io/guides/references/migration-guide#testFiles--specPattern) in Cypress 10.

The old configuration file has been migrated according to the [new changes](https://docs.cypress.io/guides/references/migration-guide#Configuration-File-Changes).

ckcherry23 reviewed (commented):
The code changes look great! 
However, some parts of the developer guide need to be updated (for example: https://reposense.org/dg/workflow.html#testing-front-end) and also the new changes from the `master` branch can be merged.

vvidday commented:
> The code changes look great! However, some parts of the developer guide need to be updated (for example: https://reposense.org/dg/workflow.html#testing-front-end) and also the new changes from the `master` branch can be merged.

Great point! Just updated the DG with the new screenshots/phrasing (the new UI is the same for v10-v12, minus the version number). Regarding the "Run all specs" portion in the DG, I removed it as the functionality was removed in Cypress 10: https://docs.cypress.io/guides/references/migration-guide#Cypress-Changes

ckcherry23 reviewed (approved):
LGTM!

HCY123902 commented on docs/dg/workflow.md:
```diff
@@ -99,9 +99,11 @@ See [vue-devtools project home](https://github.com/vuejs/vue-devtools) page for
-![Cypress Test Runner](../images/cypress-test-runner.jpg "Cypress Test Runner")
+1. Select **E2E Testing** and choose **Chrome** as the preferred browser.
+ ![Cypress Test E2E](../images/cypress-test-e2e.png)
+ ![Cypress Test Chrome](../images/cypress-test-chrome.png)
```
Can you align the indentations of the image links here

HCY123902 commented on docs/dg/workflow.md:
```diff
@@ -99,9 +99,11 @@ See [vue-devtools project home](https://github.com/vuejs/vue-devtools) page for
-![Cypress Test Runner](../images/cypress-test-runner.jpg "Cypress Test Runner")
+1. Select **E2E Testing** and choose **Chrome** as the preferred browser.
+ ![Cypress Test E2E](../images/cypress-test-e2e.png)
+ ![Cypress Test Chrome](../images/cypress-test-chrome.png)
```
Can you align the indentations of the image links here

vvidday replied:
Sorry, are you referring to the 3rd image? From my side the first 2 images look aligned:

<details>
<summary>Screenshot</summary>

![alignment](https://user-images.githubusercontent.com/45852430/224466013-dfc166c7-924a-4c7b-9408-e9a2174ad91e.png)
</details>

HCY123902 replied:
Oh I mean the indentation of the text in the file itself

vvidday replied:
Oh got it! Sorry for the misunderstanding - will fix this

vvidday replied:
> Oh I mean the indentation of the text in the file itself. The number of whitespaces is different from that of other image links here

Just noticed that the images in lines 84/86 have 3 whitespaces of indentation, but the previous links in this section had none - is there a specific one I should follow? I couldn't find anything in the style guide relating to this

HCY123902 replied:
I think you can just remove the preceding whitespaces, in the same way as how the multiline content in a list is indented

yhtMinceraft1010X reviewed (approved):
LGTM
